sroka/api/athena/athena_api_helpers.py

```diff
@@ -0,0 +1,72 @@
+import pandas as pd
+from botocore.exceptions import ClientError, EndpointConnectionError
+from retrying import retry
+
+
+def return_on_exception(filename):
+    if filename:
+        return None
+    else:
+        return pd.DataFrame([])
+
+
+@retry(stop_max_attempt_number=10,
+       wait_exponential_multiplier=1 * 1000,
+       wait_exponential_max=10 * 60 * 1000)
+def poll_status(session, _id):
+    try:
+        result = session.get_query_execution(
+            QueryExecutionId=_id
+        )
+    except ClientError as e:
+        if e.response['Error']['Code'] == 'InvalidRequestException':
+            print("Please check your query_id. Error message:")
+        else:
+            print("ClientError. Error message:")
+        print(e)
+        return None
+    except EndpointConnectionError as e:
+        print('Please check your credentials including aws_region in config.ini file and Internet connection.',
+              'Error message:')
+        print(e)
+        return None
+
+    state = result['QueryExecution']['Status']['State']
+
+    if state in ('SUCCEEDED', 'FAILED', 'CANCELLED'):
+        return result
+    else:
+        print('Waiting for {} query to end'.format(_id))
+        raise Exception
+
+
+def download_file(s3, s3_bucket, s3_key, filename):
```

Comment:
This function can be simplified:
```
def download_file(s3, s3_bucket, s3_key, filename=None):
    try:
        obj = s3.Object(s3_bucket, s3_key).get()
    except ClientError as e:
        if e.response['Error']['Code'] == 'InvalidRequestException':
            print("Please check your query. Error message:")
        else:
            print('Please check your credentials including s3_bucket in config.ini file. Error message:')
        print(e)
        return None

    if filename is None:
        df = pd.read_csv(obj['Body'])  # TODO: handle exceptions
        return df

    try:
        obj.download_file(filename)
        print('saved to {}'.format(filename))
    except FileNotFoundError as e:
        print('File or folder not found. Error message: {}'.format(e))
```

Reply:
Now I realized that it can be simplified only if `s3.Object(s3_bucket, s3_key)` checks if object exists. Otherwise there is not much to simplify ;)

Reply:
leaving it as is but I added ValueError handling for `pd.read_csv()` function